lib/contentful/request.rb

```diff
@@ -13,7 +13,12 @@ def initialize(client, endpoint, query = {}, id = nil)
 
       if id
         @type = :single
-        @id = URI.escape(id)
+        # Given the deprecation of `URI::escape` and `URI::encode`
+        # it is needed to replace it with `URI::encode_www_form_component`.
+        # This method, does replace spaces with `+` instead of `%20`.
+        # Therefore, to keep backwards compatibility, we're replacing the resulting `+`
+        # back with `%20`.
+        @id = URI.encode_www_form_component(id).gsub('+', '%20')
```

Comment:
```ruby
[4] pry(main)> URI.encode_www_form_component('this is a + sign').gsub('+','%20')
=> "this%20is%20a%20%2B%20sign"
[5] pry(main)> URI.escape 'this is a + sign'
=> "this%20is%20a%20+%20sign"
[6] pry(main)> URI.encode 'this is a + sign'
=> "this%20is%20a%20+%20sign"
````

There is a slight difference, as the `+` sign is encoded differently now. Not sure how much of an issue this will be, but I guess the %-notation should not do any harm.

Reply:
Yup, also, this is only used for sanitising IDs which do not allow `+` signs anyway. That's why I disregarded that difference.

Thanks!